lib/functions.php

```diff
@@ -752,9 +752,20 @@ function coroutine(callable $func) {
  * Upon resolution the Generator return value is used to succeed the promised result. If an
  * error occurs during coroutine resolution the returned promise fails.
  *
- * @param \Generator $generator The generator to resolve as a coroutine
+ * @param \Generator|callable $generator The generator to resolve as a coroutine
+ * @return \Amp\Promise
  */
-function resolve(\Generator $generator) {
+function resolve($generator) {
```

Comment:
Since the same var is re-used for the return value of invoking a callable, I personally think `$generator` is right